relay-server/src/actors/project_cache.rs

```diff
@@ -308,8 +309,14 @@ impl Handler<UpdateProjectState> for ProjectCache {
             })
             .into_actor(self)
             .then(move |state_result, slf, _context| {
-                let project = slf.get_or_create_project(project_key);
-                project.update_state(state_result.ok(), no_cache);
+                let state = match state_result {
+                    Ok(response) => response.state,
+                    Err(()) => Arc::new(ProjectState::err()),
```

Comment:
This is a change in behavior, triggered when project fetching reaches the deadline (maximum attempts). Previously, we reset the project state to `None`, now we fail it and drop all pending envelopes.

Reply:
Does this mean it won't be refreshed for one hour?

Reply:
It will be refreshed after the short cache timeout of 10s expires, though it can still be used for the extended grace period of up to one hour.